tests/installation/bootloader_hyperv.pm

```diff
@@ -21,57 +21,65 @@ sub hyperv_cmd {
     my ($cmd, $args) = @_;
     $args->{ignore_return_code} ||= 0;
     my $ret = console('svirt')->run_cmd($cmd);
+    diag "Command on Hyper-V returned: $ret";
     die 'Command on Hyper-V failed' unless ($args->{ignore_return_code} || !$ret);
+    return $ret;
 }
 
 sub run {
     my $svirt               = select_console('svirt');
     my $hyperv_intermediary = select_console('hyperv-intermediary');
     my $name                = $svirt->name;
 
-    # Gather paths of assets on NFS to be downloaded to Hyper-V
-    my @filepaths;
-    @filepaths = get_var('ISO') if get_var('ISO');
-    for my $n (1 .. 9) {
-        push @filepaths, "hdd\\" . basename get_var("HDD_$n") if get_var("HDD_$n");
-        push @filepaths, "iso\\" . basename get_var("ISO_$n") if get_var("ISO_$n");
-    }
-
     # Mount openQA NFS share to drive N:
     hyperv_cmd("if not exist N: ( mount \\\\openqa.suse.de\\var\\lib\\openqa\\share\\factory N: )");
 
-    # Copy assets from NFS share to local cache on Hyper-V
-    my $type = fileparse_set_fstype('MSWin32');
-    for my $filepath (@filepaths) {
-        my $filename = basename $filepath;
-        hyperv_cmd("if not exist D:\\cache\\$filename ( copy N:\\$filepath D:\\cache\\ )");
-        # Decompress the XZ compressed image & rename it to .vhd
-        if ($filename =~ m/vhdfixed\.xz/) {
-            # Becaus we wrote 30-40 GB of zeros to disk, we have to wait some time for the disk
-            # to write all the data, otherwise the test would be error-prone due to the load.
-            # Wrt `ping 127.0.0.1 ...`: this is actually the 'standard' way to sleep $nap seconds
-            # in scripted Windows environments...
-            hyperv_cmd("if not exist D:\\cache\\" . $filename =~ s/vhdfixed\.xz/vhd/r . " ( xz --decompress --keep --verbose D:\\cache\\$filename )");
-            # Check average disk load for $nap seconds to make sure all the data are in cold on disk.
-            my $nap = 600;
-            select_console('svirt');
-            type_string("typeperf \"\\PhysicalDisk(1 D:)\\Avg\. Disk Bytes/Write\"\n");
-            assert_screen('no-hyperv-disk-load', $nap);
-            send_key 'ctrl-c';
-            assert_screen 'hyperv-typeperf-command-finished';
-            select_console('hyperv-intermediary');
-            # Rename .vhdfixed to .vhd
-            $filename = $filename =~ s/\.xz//r;
-            hyperv_cmd("if not exist D:\\cache\\" . $filename =~ s/vhdfixed/vhd/r . " ( move /Y D:\\cache\\$filename D:\\cache\\" . $filename
-                  =~ s/vhdfixed/vhd/r . " )");
+    # Gather paths of assets on NFS to be downloaded to Hyper-V
+    my (@isofilepaths, @hddfilepaths);
+    @isofilepaths = get_var('ISO') if get_var('ISO');
```

Comment:
why you need this variable ? I check the code it not used anywhere

Reply:
Thanks, Anton, fixed.